right/src/usb_interfaces/usb_interface_basic_keyboard.c

```diff
@@ -205,7 +206,7 @@ void UsbBasicKeyboardSendActiveReport(void)
     UsbReportUpdateSemaphore |= 1 << USB_BASIC_KEYBOARD_INTERFACE_INDEX;
     usb_status_t status = UsbBasicKeyboardAction();
     //The semaphore has to be set before the call. Assume what happens if a bus reset happens asynchronously here. (Deadlock.)
-    if (status != kStatus_USB_Success) {
+    if (ShouldResendReport(status == kStatus_USB_Success, &retries)) {
```

Comment:
AFAICT, the `retries` counter never gets cleared. I'm not familiar enough with how the firmware works, and when it gets restarted to know whether or not this is an issue.